trillian/ctfe/services.go

```diff
@@ -82,6 +92,82 @@ func (s *issuanceChainService) Add(ctx context.Context, chain []byte) ([]byte, e
 	return hash, nil
 }
 
+func (s *issuanceChainService) FixLogLeaf(ctx context.Context, leaf *trillian.LogLeaf) error {
```

Comment:
This needs a comment explaining what it does.

Or alternatively, if you took the refactoring suggested above, then I think this method could just be applied in the storage-backed implementation of `issuanceChainService.BuildLeaf`, which is a more natural place for it. It avoids the risk that a caller will forget to call this Fixup method in any new code that needs to build a leaf.

Reply:
Added `issuanceChainService.BuildLeaf` method comment at the moment. I will see how this can be refactored together with the previous comment.

Reply:
I've refactored the `issuanceChainService.FixLogLeaf` to be called immediately after the Trilian gRPC call and wrapped them into another method.